google/cloud/spanner_dbapi/connection.py

```diff
@@ -369,14 +382,30 @@ def close(self):
         The connection will be unusable from this point forward. If the
         connection has an active transaction, it will be rolled back.
         """
-        if self.inside_transaction:
+        if self.spanner_transaction_started:
             self._transaction.rollback()
```

Comment:
Yes, self._transaction would be None so spanner_transaction_started would be False. Wouldn't that be correct?

Reply:
I don't think it's correct that `spanner_transaction_started` would be false in all cases. If `readonly=true` and you execute `BEGIN`, then that is the state when you come here?

Reply:
Currently `spanner_transaction_started` is only true when ReadWrite Transaction is in progress. It would work here but I can see session leaks in commit for RO transaction. 
I guess we need to have 2 properties `spanner_rw_transaction_started` and `spanner_ro_transaction_started` and build logic based on this at various places. WDYT?

Reply:
I don't think we need two different flags. I think that we can achieve the same goal with just one flag, and then check whether `_transaction` or `_snapshot` is not None.

Reply:
Done